sdk/keyvault/azure-security-keyvault-administration/src/samples/java/com.azure.security.keyvault.administration/CreateRoleAssignmentsForDifferentScopes.java

```diff
@@ -0,0 +1,58 @@
+import com.azure.security.keyvault.administration.models.KeyVaultRoleAssignmentScope;
+
+/**
+ * This sample demonstrates how to create role assignments in the key vault for different scopes.
+ */
+public class CreateRoleAssignmentsForDifferentScopes {
+    /**
+     * Authenticates with the key vault and shows how to create role assignments in the key vault for different scopes.
+     *
+     * @param args Unused. Arguments to the program.
+     * @throws IllegalArgumentException when an invalid key vault URL is passed.
+     */
+    public static void main(String[] args) {
+        // Instantiate an KeyVaultAccessControlClient that will be used to call the service. Notice that the client is
+        // using default Azure credentials. To make default credentials work, ensure that environment variables
+        // 'AZURE_CLIENT_ID', 'AZURE_CLIENT_KEY' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+        KeyVaultAccessControlClient accessControlClient = new KeyVaultAccessControlClientBuilder()
+            .vaultUrl("https://{YOUR_VAULT_NAME}.vault.azure.net")
+            .credential(new DefaultAzureCredentialBuilder().build())
+            .buildClient();
+
+        // By default role assignments apply to the global scope. It is also possible to be more specific by applying an
+        // assignment to the all keys scope or a specific KeyVaultKey.
+
+        // Let's assign a role to a service principal so that it applies to all keys. To do this we'll need a service
+        // principal object ID and a role definition ID. A role definition ID can be obtained from the 'id' property of
+        // one of the role definitions returned from listRoleAssignments().
+        String roleDefinitionId = "<role-definition-id>";
+        String servicePrincipalId = "<service-principal-id>";
+
+        KeyVaultRoleAssignmentProperties roleAssignmentProperties =
+            new KeyVaultRoleAssignmentProperties(roleDefinitionId, servicePrincipalId);
+
+        KeyVaultRoleAssignment roleAssignmentForAllKeys =
+            accessControlClient.createRoleAssignment(KeyVaultRoleAssignmentScope.GLOBAL, roleAssignmentProperties);
+
+        System.out.printf("Created role assignment with name: %s %n", roleAssignmentForAllKeys.getName());
+
+        // Now let's assign a role to a service principal so that it applies to a specific KeyVaultKey. To do this we'll
+        // use the role definition ID and a service principal object ID from the previous sample. We'll also need the
+        // ID of an existing KeyVaultKey, which can be obtained from the service using a KeyClient.
+        String keyId = "<key-id>";
+
+        KeyVaultRoleAssignment roleAssignmentForSingleKey =
+            accessControlClient.createRoleAssignment(KeyVaultRoleAssignmentScope.fromString(keyId),
+                roleAssignmentProperties);
+
+        System.out.printf("Created role assignment with name: %s %n", roleAssignmentForSingleKey.getName());
+    }
+}
```

Comment:
No async equivalent? Probably fine, but just calling it out since the other examples had an async version. Seems to be the case below as well. If you feel it has diminished returns at this point, that's understandable. A couple good ones should show the idea.

Reply:
It doesn't take much effort to add an async sample, I guess I thought it wasn't necessary since we have the hello worlds. I can add that it before we merge :)

Reply:
Ended expanding the samples for the scenario where we resume backup and restore operations and also added an async counterpart.